main.cc

```diff
@@ -1460,6 +1460,10 @@ int main(int ac, char** av) {
                 service::get_local_storage_service().drain_on_shutdown().get();
             });
 
+            auto stop_sl_updates = defer_verbose_shutdown("service level controller updates", [] {
+                sl_controller.local().drain().get();
+            });
+
```

Comment:
If someone does `nodetool drain` and then stops scylla, this deferred action will be called in the condition where commitlog is down and migration manager is stoped, i.e. as if it was called _after_ `drain_on_shutdown`. Is that OK?

Reply:
It is now, because the updater only reads from Scylla, it doesn't need to write anything.

Reply:
OK, but why is `s_l_controller::drain` put to be run before `storage_service::drain_on_shutdown`?

Reply:
Actually there's no reason now. I'll put it after, to match the behavior if somebody runs `nodetool drain`